### PR TITLE
Add types for nodejs sdk

### LINKED ISSUE #2937: TypeScript types for Node SDK
It would be rad to bundle types with the `@google-cloud/agones-sdk` NPM package :+1:

### PR DESCRIPTION
**Pull request by vasily-polonsky**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug
> /kind cleanup
> /kind documentation

/kind feature
> /kind hotfix

**What this PR does / Why we need it**:
In npm package there are no types. Should be helpful for typescript projects.

**Which issue(s) this PR fixes**:

Closes #2937

**Special notes for your reviewer**:
Many return types are not perfect, but I dont know where to get the proper types, that can be used here.
One more thing, I'm not sure if this is enough to get the types into the npm package or if some other changes are needed also

### CODE REVIEW

**Comment by steven-supersolid** [comment]
Nice, this will be really useful. Will check the types match

File: sdks/nodejs/src/agonesSDK.d.ts
Comment (steven-supersolid) [inline sdks/nodejs/src/agonesSDK.d.ts:30]: When using TS I usually use `Record<string, unknown>` here as would cast the type to something useful but maybe `any` is more flexible?
We could also supply full return types, e.g. interfaces/types for all these, or better to do in a separate PR?

Comment (vasily-polonsky) [reply sdks/nodejs/src/agonesSDK.d.ts]: As for `unknown` and `any`, `unknown` is definitely better when you can distinguish type, there's no `instanceof` type check you can do, or another field like `isAxiosError` to distinguish a particular type. The problem I see with `unknown` is that in this particular case, you have to write
```
if(typeof awaitedValue === 'object') {
  if(typeof awaitedValue.prop1 === 'object') {
    awaitedValue.prop1.prop1A
  }
}
```
with `any` you can just use awaitedValue?prop1?prop1A with the same security.
Maybe I am missing something?

Comment (vasily-polonsky) [reply sdks/nodejs/src/agonesSDK.d.ts]: Can we provide full return types? That would be much better. I just started working with Agones last week, but for me the real result only returns for `getGameServer` and `watchGameServer` methods, for the rest sdk returns an empty object, maybe something is wrong in my cluster.
if we could specify the exact types, I would definitely do in this PR

**Comment by markmandel** [comment]
@steven-supersolid lemme know when this is all good, and happy to approve and merge 👍🏻

**Review by steven-supersolid — APPROVED** [review]
LGTM. Great stuff!

**Comment by vasily-polonsky** [comment]
Sorry, read the pull request process, and thought it should be merged automatically after approval, but it wasn't. Thought it was because of commits ahead in the main branch, merged the main branch here, but that action removed the approval. What should I do?

**Comment by steven-supersolid** [comment]
Still LGTM

**Comment by zmerlynn** [comment]
> Sorry, read the pull request process, and thought it should be merged automatically after approval, but it wasn't. Thought it was because of commits ahead in the main branch, merged the main branch here, but that action removed the approval. What should I do?

Tag an approver and we can handle it. I'll do so now!

**Review by zmerlynn — APPROVED** [review]

**Pull request merged**
